catroid/src/main/java/org/catrobat/catroid/content/actions/AskAction.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.actions
+
+import android.util.Log
+import com.badlogic.gdx.scenes.scene2d.Action
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.InterpretationException
+import org.catrobat.catroid.formulaeditor.UserVariable
+import org.catrobat.catroid.stage.StageActivity
+import java.util.ArrayList
+
+class AskAction : Action() {
+    private var sprite: Sprite? = null
+    private var questionFormula: Formula? = null
+    private var answerVariable: UserVariable? = null
+    private var questionAsked = false
+    private var answerReceived = false
+    private fun askQuestion() {
+        if (StageActivity.messageHandler == null) {
+            return
+        }
+        var question = ""
+        try {
+            if (questionFormula != null) {
+                question = questionFormula!!.interpretString(sprite)
+            }
```

Comment:
```suggestion
            question = questionFormula?.interpretString(sprite)
```

Reply:
Null checks like this are really simple in Kotlin

Reply:
You could also just do ``
```suggestion
            question = questionFormula?.interpretString(sprite)
```
since the question will be null otherwise anyway